modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -836,6 +868,101 @@
   (let [table-str (get-table-str table)]
     [(str "DROP TABLE IF EXISTS " table-str)]))
 
+(defmethod driver/create-table! :bigquery-cloud-sdk
+  [driver database-id table-name column-definitions & {:keys [primary-key]}]
+  (let [sql (#'driver.sql-jdbc/create-table!-sql driver table-name column-definitions :primary-key primary-key)]
+    (driver/execute-raw-queries! driver (t2/select-one :model/Database database-id) [sql])))
+
+(defmethod driver/rename-tables!* :bigquery-cloud-sdk
+  [driver db-id sorted-rename-map]
+  (let [database (t2/select-one :model/Database db-id)]
+    (doseq [[old-table-name new-table-name] sorted-rename-map]
+      (let [old-table-str (get-table-str old-table-name)
+            new-table-str (name new-table-name)]
+        (try
+          (let [sql (format "ALTER TABLE %s RENAME TO %s" old-table-str new-table-str)]
+            (driver/execute-raw-queries! driver database [sql]))
+          (catch Exception e
+            (if (and (instance? com.google.cloud.bigquery.BigQueryException e)
+                     (str/includes? (.getMessage e) "streaming data"))
+              ;; rename failed, we must create + drop
```

Comment:
This is one of those 'the contract is not clear' type things I suppose:

- Is rename-tables* expected to 'rename at any cost' or have an expected performance/locking/blocking profile?
- I also do not think this holds for bigquery now:
  >  Implementations should perform the actual rename operations atomically as supported by the database.

Reply:
yeah i failed to anticipate databases not supporting rename. this is OK for now but let's follow up [soon](https://linear.app/metabase/issue/QUE-2474/improve-error-handling-around-table-swap)